tests/lambda.test

```diff
@@ -0,0 +1,27 @@
+// This is an LLVM-style lit test that uses FileCheck.
+// The test checks that unused generated lambda operators (i.e. members of the
+// generated "closure type" class) are not reported as unused.
+//
+// RUN: rm -rf %t && mkdir -p %t
+// RUN: split-file %s %t
+// RUN: sed -e "s|TESTDIR|%t|g" %t/compile_commands.json.in > %t/compile_commands.json
+// RUN: xunused %t/compile_commands.json 2>&1 | FileCheck %s
+// CHECK-NOT: {{.*}}operator(){{.*}}unused
```

Comment:
We are using a regex here to avoid making this negative check dependent on exact line numbers (for generated functions!) and other potentially unstable/non-standardized details and, hence, silently failing to detect regressions when those details change.

Many existing tests suffer from the same problem (among others). We are working on a comprehensive solution that will convert these fragile negative tests into reliable positive ones, but that work deserves a dedicated PR. Until then, the regex proposed in this PR should work OK.

@mgehre, if you prefer, this regex can be replaced with an exact error string, including a line number. This will make this test more similar to existing tests (but also more fragile).